spikeinterface/sortingcomponents/matching/wobble.py

```diff
@@ -0,0 +1,948 @@
+        return kwargs
+
+    @classmethod
+    def get_margin(cls, recording, kwargs):
+        """Get margin for chunking recording.
+
+        Parameters
+        ----------
+        recording : RecordingExtractor
+            The recording extractor object.
+        kwargs : dict
+            Keyword arguments for matching method.
+
+        Returns
+        -------
+        margin : int
+            Buffer in samples on each side of a chunk.
+        """
+        buffer_ms = 10
+        # margin = int(buffer_ms*1e-3 * recording.sampling_frequency)
+        margin = 300 # To ensure equivalence with spike-psvae version of the algorithm
+        return margin
+
+    @classmethod
+    def main_function(cls, traces, method_kwargs):
+        """Main function that performs template matching.
+
+        Includes spike time correction to ensure spike times are at trough, and unit id correction to account for
+        upsampling.
+
+        Parameters
+        ----------
+        traces : ndarray (chunk_len + 2*margin, n_chan)
+            Voltage traces for a chunk of the recording.
+        method_kwargs : dict
+            Keyword arguments for matching method.
+
+        Returns
+        -------
+        spikes : ndarray (n_spikes,)
+            Resulting spike train.
+        """
+        # Unpack method_kwargs
+        nbefore, nafter = method_kwargs['nbefore'], method_kwargs['nafter']
+        template_meta = method_kwargs['template_meta']
+        params = method_kwargs['params']
+        sparsity = method_kwargs['sparsity']
+        objective = method_kwargs['objective']
+
+        # Check traces
+        traces = traces.astype(np.float32, casting='safe') # ensure traces are specified as np.float32
+        objective.obj_len = get_convolution_len(traces.shape[0], template_meta.n_time)
+
+        # run using run_array
+        spike_train, scalings, distance_metric = cls.run_array(traces, template_meta, params, sparsity, objective)
+
+        # extract spiketrain and perform adjustments
+        spike_train[:, 0] += nbefore
+        spike_train[:, 1] //= params.jitter_factor
+
+        # TODO : Benchmark spike amplitudes
+        # Find spike amplitudes / channels
+        amplitudes, channel_inds = [], []
+        for i, spike_idx in enumerate(spike_train[:, 0]):
+            best_ch = np.argmax(np.abs(traces[spike_idx, :]))
+            amp = np.abs(traces[spike_idx, best_ch])
+            amplitudes.append(amp)
+            channel_inds.append(best_ch)
+
+        # assign result to spikes array
+        spikes = np.zeros(spike_train.shape[0], dtype=cls.spike_dtype)
+        spikes['sample_ind'] = spike_train[:, 0]
+        spikes['cluster_ind'] = spike_train[:, 1]
+        spikes['channel_ind'] = channel_inds
+        spikes['amplitude'] = amplitudes
+
+        return spikes
+
+
+    @classmethod
+    def run_array(cls, traces, template_meta, params, sparsity, objective):
```

Comment:
``run_array`` was a legacy name from the spike_psvae repo.  I decided to just unpack the whole function into `main_function`, which I think works pretty well.

The scalings and distance metric are useful properties for each spike, so I don't want to just get rid of them.  I think we should add them as attributes to the ``spikes`` output array, but it could be nice to discuss.